sekoia_automation/connector/__init__.py

```diff
@@ -285,20 +313,84 @@ def _chunk_events(self, events: Sequence) -> Generator[list[Any], None, None]:
 
         # if events were discarded, log it
         if nb_discarded_events > 0:
+            self._discarded_events.labels(intake_key=self.configuration.intake_key).inc(
+                nb_discarded_events
+            )
+
             self.log(
                 message=f"{nb_discarded_events} too long events "
                 "were discarded (length > 250kb)"
             )
 
-    def forward_events(self, events) -> None:
+    def forward_events(self, events: Sequence[EventType]) -> None:
         try:
             chunks = self._chunk_events(events)
             _name = self.name or ""  # mypy complains about NoneType in annotation
             for records in chunks:
                 self.log(message=f"Forwarding {len(records)} records", level="info")
                 self.send_records(
                     records=list(records),
-                    event_name=f"{_name.lower().replace(' ', '-')}_{time()!s}",
+                    event_name=f"{_name.lower().replace(' ', '-')}_{datetime_time()!s}",
                 )
         except Exception as ex:
             self.log_exception(ex, message="Failed to forward events")
+
+    def iterate(self) -> Generator[tuple[list[EventType], datetime | None], None]:
+        """Iterate over events."""
+        yield [], None
```

Comment:
Maybe throw NotImplementedError to make it clear that this function has to be implemented?
```suggestion
        raise NotImplementedError
```